lib/Doctrine/ORM/Query/ParameterTypeInferer.php

```diff
@@ -53,7 +53,7 @@ public static function inferType($value)
             return Type::BOOLEAN;
         }
 
-        if ($value instanceof \DateTime) {
+        if ($value instanceof \DateTime || $value instanceof \DateTimeImmutable) {
```

Comment:
Undefined class in instanceof check is not an error, is it?
Well you can still make it like this, if you wanna be pedantic:
```php
if ($value instanceof \DateTime || (PHP_VERSION_ID >= 50500 && $value instanceof \DateTimeImmutable)) {
```